### PR TITLE
Fix/upgrade-cratis

### PR DESCRIPTION
**Pull request by einari**
### Fixed

- Nginx config now works with WebSockets again. The headers was placed in the wrong section (`server`) while it should be on the `location` sections.
- Upgraded to Cratis 9 and the new Aksio Applications and MongoDB libraries.
- Updated how we configure the Cratis 9 client after its now configured from `AppSettings.json` rather than its own custom config file.

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<Project>
+	<PropertyGroup>
```
Comment (Torgeir-Hansen) [inline Directory.Packages.props:2]: Haven't decided on spaces or tabs yet? ;)

Comment (einari) [reply Directory.Packages.props:2]: Good catch. We don't want to end up like this: https://www.youtube.com/watch?v=SsoOG6ZeyUI

File: Source/Reactions/Applications/ApplicationEnvironmentResourcesCoordinator.cs [hunk 2]
```diff
@@ -77,12 +77,14 @@ public async Task AppSettingsChanged(AppSettingsSetForMicroservice @event, Event
...
-
         var subscription = _settings.AzureSubscriptions.First(_ => _.SubscriptionId == environment.AzureSubscriptionId);
         var storage = await application.GetStorage(environment, _settings.ServicePrincipal, subscription);
         var fileStorage = new FileStorage(storage.AccountName, storage.AccountKey, microservice.GetFileShareName(), _fileStorageLogger);
         var microserviceStorage = new MicroserviceStorage(application, microservice, fileStorage);
-        microserviceStorage.CreateAndUploadAppSettings();
+
+        // TODO: Need to store information to be able to set the Azure Storage account connection string for AppSettings
+        // microserviceStorage.CreateAndUploadAppSettings();
+        throw new NotImplementedException();
```
Comment (Torgeir-Hansen) [inline Source/Reactions/Applications/ApplicationEnvironmentResourcesCoordinator.cs:87]: NotImplemented?

Comment (einari) [reply Source/Reactions/Applications/ApplicationEnvironmentResourcesCoordinator.cs:87]: Deliberate - if you look at the ToDo above it. Will revisit this.

File: Source/Reactions/Reactions.csproj [hunk 3]
```diff
@@ -7,12 +7,12 @@
 		<ProjectReference Include="../Infrastructure/Infrastructure.csproj" />
 	</ItemGroup>
 	<ItemGroup>
-		<PackageReference Include="handlebars.net" Version="2.1.2" />
-		<PackageReference Include="Azure.Storage.Files.Shares" Version="12.12.1" />
-		<PackageReference Include="Azure.Storage.Blobs" Version="12.14.1" />
-		<PackageReference Include="Microsoft.Azure.Management.Fluent" Version="1.38.1" />
-		<PackageReference Include="Microsoft.Azure.Management.ResourceManager.Fluent" Version="1.38.1" />
-		<PackageReference Include="semver" Version="2.2.0" />
+		<PackageReference Include="handlebars.net" />
```
Comment (Torgeir-Hansen) [inline Source/Reactions/Reactions.csproj:10]: Uten versjoner? Det vil jo bare fungere frem til ny pakke er tilgjengelig - og vips så slutter potensielt bygg og/eller runtime å fungere uten at vi aner hvorfor

File: Source/Bootstrap/Bootstrap.csproj [hunk 4]
```diff
@@ -10,7 +10,7 @@
     </ItemGroup>
 
     <ItemGroup>
-      <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0" />
+      <PackageReference Include="Microsoft.Extensions.Logging.Console" />
```
Comment (Torgeir-Hansen) [inline Source/Bootstrap/Bootstrap.csproj:13]: Uten versjon? Det vil jo bare fungere frem til ny pakke er tilgjengelig - og vips så slutter potensielt bygg og/eller runtime å fungere uten at vi aner hvorfor

File: Source/Reactions/Applications/Pulumi/ApplicationIngressPulumiExtensions.cs [hunk 5]
```diff
@@ -410,47 +413,49 @@ static IEnumerable<Domain> GetAllDomains(ApplicationEnvironmentWithArtifacts env
 
             case IdentityProviderType.IdPorten:
                 var proxyAuthorizationEndpoint = $"https://{ingress.AuthDomain!.Name}/.aksio/id-porten/authorize";
-                var client = new HttpClient();
-                var url = $"{identityProvider.Issuer}/.well-known/openid-configuration";
-                var result = await client.GetAsync(url);
-                var json = await result.Content.ReadAsStringAsync();
+                using (var client = new HttpClient())
```
[the remark this answers is not in the document]
Comment (einari) [reply Source/Reactions/Applications/Pulumi/ApplicationIngressPulumiExtensions.cs:416]: Not supported in switch / case statements.

File: Source/Bootstrap/Bootstrap.csproj (hunk 4 above)
Comment (einari) [reply Source/Bootstrap/Bootstrap.csproj:13]: Directory.Packages.props in the root sets all versions

File: Source/Infrastructure/Infrastructure.csproj [hunk 6]
```diff
@@ -3,10 +3,10 @@
         <ProjectReference Include="../Concepts/Concepts.csproj" />
     </ItemGroup>
 	<ItemGroup>
-		<PackageReference Include="pulumi.automation" Version="3.54.1" />
-		<PackageReference Include="pulumi.azuread" Version="5.35.0" />
-		<PackageReference Include="pulumi.azurenative" Version="1.103.0" />
-		<PackageReference Include="Pulumi.Mongodbatlas" Version="3.7.0" />
-        <PackageReference Include="System.Text.Encoding.Extensions" Version="4.3.0" />
+		<PackageReference Include="pulumi.automation" />
```
[the remark this answers is not in the document]
Comment (einari) [reply Source/Infrastructure/Infrastructure.csproj:6]: Directory.Packages.props in the root sets all versions

File: Source/Reactions/Reactions.csproj (hunk 3 above)
Comment (einari) [reply Source/Reactions/Reactions.csproj:10]: Directory.Packages.props in the root sets all versions

**Review by Torgeir-Hansen — APPROVED** [review]

**Pull request merged**